### PR TITLE
Retry test run status checks

### PR DESCRIPTION
**Pull request by jonstoneman**
Retry test run status checks

This will allow us to recover from transient network issues

### CODE REVIEW

File: src/commands/test/lib/state-checker.ts [hunk 1]
```diff
@@ -28,12 +28,29 @@ export class StateChecker {
...
+      try {
+        state = await out.progress("Checking status...", this.getTestRunState(this.client, this.testRunId));
+      } catch (error) {
+        errorCount++;
+        if (errorCount >= maxErrors) {
+          throw error;
+        }
+
+        await out.progress("Status check failed, retrying...", this.delay(1000 * errorRetryWait));
+        continue;
```
Comment (jerietve) [inline src/commands/test/lib/state-checker.ts]: This shortcuts the existing timeout behavior. Please make sure the method still exits after `timeoutSec` seconds.

**Comment by jpeg-ms** [comment]
At a quick glance, this looks fine, but was it tested? If yes, then how? (just curious there)

**Comment by jonstoneman** [comment]
I started test runs and disabled wifi - it showed the status check retries occurred and eventually errored correctly. If I re-enabled wifi before the retry limit was hit then the status checks started working again.

I also added temporary code to set timeoutSec to a low number and confirmed that the timeout works with and without wifi disabled.

**Comment by jonstoneman** [comment]
Is there a release planned or could we get one for this change?

**Comment by jerietve** [comment]
@joabalea and @jpeg-ms are planning a release today. You can check the Slack channel for details.

**Comment by joabalea** [comment]
@jonstoneman We just released v2.3.1

**Comment by jonstoneman** [comment]
Thanks!

**Pull request merged**